### PR TITLE
Code quality: Refactoring and improvements 

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Related #9895 
- Related #10359
- Related #4180
- Related #10268 

**Details**
This PR contains a partial refactoring of our code base. Here's the change-list:
- Removed nullable warnings (verifying that variables are null checked)
- Removed unused parameters from methods
- Removed some avoidable `if-else` branches
- Created methods for code that can be reused

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.App/UserControls/Widgets/DrivesWidget.xaml.cs [hunk 1]
```diff
@@ -155,7 +153,7 @@ private async void OpenInNewTab_Click(object sender, RoutedEventArgs e)
 		private async void OpenInNewWindow_Click(object sender, RoutedEventArgs e)
 		{
 			var item = ((MenuFlyoutItem)sender).DataContext as DriveItem;
-			if (await DriveHelpers.CheckEmptyDrive(item?.Path))
+			if (await DriveHelpers.CheckEmptyDrive(item?.Path) || item is null)
```
Comment (gave92) [inline src/Files.App/UserControls/Widgets/DrivesWidget.xaml.cs:156]: `if (item is null || await DriveHelpers.CheckEmptyDrive(item.Path))`

File: src/Files.App/ViewModels/ItemViewModel.cs [hunk 2]
```diff
@@ -386,53 +374,48 @@ public ItemViewModel(FolderSettingsViewModel folderSettingsViewModel)
...
-				{
-					await AddFileOrFolderAsync(newListedItem);
-					await OrderFilesAndFoldersAsync();
-					await ApplySingleFileChangeAsync(newListedItem);
-				}
-			}
+			if (!CommonPaths.RecycleBinPath.Equals(CurrentFolder?.ItemPath, StringComparison.OrdinalIgnoreCase))
+				return;
+			using var folderItem = SafetyExtensions.IgnoreExceptions(() => new ShellItem(e.FullPath));
+			if (folderItem is null) return;
```
Comment (QuaintMako) [inline src/Files.App/ViewModels/ItemViewModel.cs:408]: Return should be on another line.

**Review by QuaintMako — DISMISSED** [review]

**Review by hecksmosis — APPROVED** [review]

File: src/Files.App/UserControls/Widgets/DrivesWidget.xaml.cs [hunk 3]
```diff
@@ -156,49 +147,39 @@ private async void EjectDevice_Click(object sender, RoutedEventArgs e)
...
-			await NavigationHelpers.OpenPathInNewTab(item.Path);
+			await NavigationHelpers.OpenPathInNewTab(item?.Path);
 		}
 
 		private async void OpenInNewWindow_Click(object sender, RoutedEventArgs e)
 		{
 			var item = ((MenuFlyoutItem)sender).DataContext as DriveItem;
-			if (await CheckEmptyDrive(item.Path))
-			{
+			if (item is null || await DriveHelpers.CheckEmptyDrive(item.Path))
```
Comment (QuaintMako) [inline src/Files.App/UserControls/Widgets/DrivesWidget.xaml.cs:158]: `if (await DriveHelpers.CheckEmptyDrive(item?.Path))` to be in line with the other tests around.

Comment (ferrariofilippo) [reply src/Files.App/UserControls/Widgets/DrivesWidget.xaml.cs]: Done. Note that I had to edit `NavigationHelpers.OpenPathInNewWindowAsync()`

**Review by hecksmosis — APPROVED** [review]
LGTM

**Review by QuaintMako — APPROVED** [review]

**Comment by yaira2** [comment]
@ferrariofilippo can you resolve the merge conflicts?

**Review by hecksmosis — APPROVED** [review]

**Comment by d2dyno1** [comment]
One more conflict :eyes:

**Review by hecksmosis — APPROVED** [review]

**Review by d2dyno1 — APPROVED** [review]

**Pull request merged**